### PR TITLE
Add useStableQuery.ts

### PR DESCRIPTION
**Pull request by vakila**
Contains `useStableQuery` and `useStablePaginatedQuery` hooks detailed in upcoming Stack post

Feedback more than welcome!

### CODE REVIEW

**Comment by vakila** [comment]
cc @ianmacartney

File: src/hooks/useStableQuery.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+/**
+ * Drop-in replacement for usePaginatedQuery for use with a parametrized query.
+ * Unlike usePaginatedQuery, when query arguments change useStablePaginatedQuery 
+ * does not return empty results and 'LoadingMore' status. Instead, it continues
+ * to return the previously loaded results until the new results have finished
+ * loading.
+ *
+ * See stack.convex.dev post "Help, my data is overreacting!" for details.
```
Comment (ianmacartney) [inline src/hooks/useStableQuery.ts:38]: You can point at https://stack.convex.dev/help-my-app-is-overreacting btw - it'll be available once I publish it, but not on the homepage until after the specified date

File: src/hooks/useStableQuery.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  // result is only undefined while data is loading
+  // if a freshly loaded result is available, use the ref to store it
+  if (result !== undefined) {
+    stored.current = result
+  }
+
+  // undefined on first load, stale data while loading, fresh data after loading
+  return stored.current 
+}) as typeof useQuery
```
Comment (ianmacartney) [inline src/hooks/useStableQuery.ts:28]: wow TIL. This is so much easier than typing it all out

**Review by ianmacartney — APPROVED** [review]

**Comment by vakila** [comment]
Ah sorry I missed your comments on Friday, Ian! Only saw the PR accepted notification come through, not the other ones - I'll submit a new PR to add the link, thanks!

**Pull request merged**